pyro/infer/mcmc/hmc.py

```diff
@@ -160,6 +186,62 @@ def _find_reasonable_step_size(self, z):
             direction_new = 1 if target_accept_logprob < -delta_energy else -1
         return step_size
 
+    def _configure_adaptation(self, prototype_trace):
+        self._adapt_phase = True
+        # Setup up the schedule for parameter adaptation.
+        self._adaptation_schedule = {0: adaptation_window(adapt_step_size=self.adapt_step_size, adapt_mass=False)}
+        if self.adapt_step_size:
+            z = {name: node["value"] for name, node in prototype_trace.iter_stochastic_nodes()}
+            for name, transform in self.transforms.items():
+                z[name] = transform(z[name])
+            self.step_size = self._find_reasonable_step_size(z)
+            self.num_steps = max(1, int(self.trajectory_length / self.step_size))
+            # make prox-center for Dual Averaging scheme
+            loc = math.log(10 * self.step_size)
+            self._adapted_scheme = DualAveraging(prox_center=loc)
+        if self.adapt_mass:
+            if self.mass_structure == 'identity':
+                self._mass_estimator = None
+            elif self.mass_structure == 'diagonal':
+                self._mass_estimator = WelfordCovariance(diagonal=True)
+            elif self.mass_structure == 'dense':
+                self._mass_estimator = WelfordCovariance(diagonal=False)
+            else:
+                raise ValueError(
+                        "mass_type must be one of: 'identity', 'diagonal', 'dense' (got '{}')".format(
+                                self.mass_structure))
+            # Buffer and window sizes from stan
+            start_buffer = 75
+            end_buffer = 25
+            initial_window = 50
+            total = start_buffer + end_buffer + initial_window
+            if self._warmup_steps < total:
+                raise ValueError(
```

Comment:
How about we just use these number of warmup steps by default if step size or mass adaptation is enabled?